src/metabase/models/table.clj

```diff
@@ -296,3 +297,117 @@
+
+   :joins        {:db [:model/Database [:= :db.id :this.db_id]]}})
+
+(search/define-spec "segment"
+  {:model        :model/Segment
+   :context      {:collection_id :id
+                  :database_id   :table.db_id
+                  :table_id      :table_id}
+   :archived     :archived
+   :search-terms [:name :description]
+
+   :render-terms [[:table.description :table_description]
+                  [:table.schema :table_schema]
+                  [:table.name :table_name]]
+
+   :created-at   true
+   :updated-at   true
+
+   :joins        {:table [:model/Table [:= :table.id :this.table_id]]}})
+
+(search/define-spec "collection"
+  {:model        :model/Collection
+   :context      {:collection_id :id
+                  :database_id   nil
+                  :table_id      nil}
+   :archived     :archived
+   :search-terms [:name]
+
+   :render-terms [[:type :collection_type]
+                  [:name :collection_name]
+                  [:authority_level :collection_authority_level]
+                  :archived_directly
+                  ;; why not make this a search term?
+                  :description
+                  ;; bookmark (inserted just-in-time from join for ranking and rendering)
+                  :location]
+
+   :created-at   true
+   :updated-at   false
+
+   :skip         {:namespace [:not nil]}
+
+   ;; depends on the current user, used for rendering and ranking
+   :bookmark     [:model/CollectionBookmark [:and
+                                             [:= :bookmark.collection_id :this.id]
+                                             ;; a magical alias, :current_user (or perhaps this clause can be implicit)
+                                             [:= :bookmark.user_id :current_user/id]]]})
+
+(search/define-spec "card"
+  {:model        :model/Card
+   :context      {:collection_id :collection_id
+                  :database_id   nil
+                  :table_id      nil}
+   :archived     :archived
+   ;; TODO :dataset_query if native
+   :search-terms [:name :description]
+
+   :render-terms [[:collection.name :collection_name]
+                  [:collection.type :collection_type]
+                  [:collection.location :collection_location]
+                  [:collection.authority_level :collection_authority_level]
+                  :archived_directly
+                  :collection_position
+                  :creator_id
+                  ;; dashboard count (TODO oh, best query this when rendering)
+                  [:r.timestamp :last_edited_at]
+                  [:r.user_id :last_editor_id]
+                  [:mr.status :moderated_status]
+                  :display
+                  :dataset_query]
+
+   :bookmark     [:model/CardBookmark [:and
+                                       [:= :bookmark.card_id :this.id]
+                                       [:= :bookmark.user_id :current_user/id]]]
+
+   :skip         {:this.type            [:not "question"]
```

Comment:
I think this case is better defined as `:= :this.type :question` rather than `:not= :this.type :question`. right now we only have 2 types but this will return wrong data if we have more.

also, wdyt instead of `skip` we use `where` directly? I think we're quite familiar with honeysql so it's actually easier to keep this close to it.

Reply:
Yeah I started with `:where`, flipped it at some point, and now regret that. Depending on which model it is its very subjective which is more straightforward, and you're right this case is very awkward, so might as well keep the implementation less "novel".

The thing that's really convinced me is for the various kinds of questions, I want to be able just transform a "base" map, and `sql.helpers/where` will "just work" if I flip things back around.

Reply:
Worked out nicely